client/app/assets/less/inc/base.less

```diff
@@ -261,12 +261,6 @@ text.slicetext {
       favorites-control {
         float: left;
       }
-
-      h3 {
-        width: 100%;
-        margin-bottom: 5px !important;
-        display: block !important;
-      }
```

Comment:
I noticed some difference between old and new header version: when query does not have tags, tags control should stay on the same line with query name (see screenshots). Seems this style is related to that

![image](https://user-images.githubusercontent.com/12139186/70648068-f914b000-1c52-11ea-80a0-c12080fafea7.png)
![image](https://user-images.githubusercontent.com/12139186/70648154-25303100-1c53-11ea-8874-fe1016af4928.png)

Reply:
Seems it's better to keep markup from base branch, unless you have some better UX ideas for this component. Also, if you think that button group does not fit here - feel free to remove it.

Reply:
It wraps in the older version too, just take more for it to happen :slightly_smiling_face:. That's due to the existing conflict with `.page-title { display: block }`, fixed in 37c9b469cad6e0b245e31fcc1be4fb5176bffeeb

Reply:
Now layout is good, but still there some differences:

existing
![image](https://user-images.githubusercontent.com/12139186/70651840-40526f00-1c5a-11ea-98c3-e6e6e68de818.png)

react
![image](https://user-images.githubusercontent.com/12139186/70651858-4c3e3100-1c5a-11ea-93b0-a0b8fc1f1ea5.png)

Notice: favorites control should be on the same line with query name, too much space between query name and tags, and tags should go under buttons on right side.

Reply:
Yeah, in the original there are two `QueryTagsControl` components, one for the desktop position and another for the mobile one :sleepy:

Reply:
Decided to come back with the old markup and move on for now

Reply:
Agreed. And I think - what if we'll use some HOC (?) to conditionally change a markup depending on viewport width (similar to what dashboard grid does)?

Reply:
I usually feel css is more reliable than js when it can be used, do you have any advantages in mind?

Reply:
In this case the only advantage is that tags component will be rendered only once (+markup will simplified a bit). CSS Grid is the perfect solution, but it still is not supported good enough. For JS, there is a [`matchMedia`](https://developer.mozilla.org/en-US/docs/Web/API/Window/matchMedia) API which is 100% supported in all our target browsers.

Reply:
BTW you meant to use a hook, right? (like [this](https://usehooks.com/useMedia/) one)

It's an interesting hook to have anyway, I tried to use it once I wanted to check in JS if the browser was in Percy,

Reply:
Yes, something like that :slightly_smiling_face:

Reply:
That hook is not hard to be applied in this context:
```jsx
const isMobile = useMedia(['(max-width: 880px)'], [true], false);
```

But I prefer keeping it in css for now as I don't want to break the angular page yet (we can reconsider when cleaning angular stuff)